File: clients/Python-PyPy/model/debug_state.py
from .vec2_int import Vec2Int
from .vec2_float import Vec2Float
from .vec2_float import Vec2Float
from .camera import Camera
class DebugState:
    def __init__(self, window_size, mouse_pos_window, mouse_pos_world, pressed_keys, camera, player_index):
        self.window_size = window_size
        self.mouse_pos_window = mouse_pos_window
        self.mouse_pos_world = mouse_pos_world
        self.pressed_keys = pressed_keys
        self.camera = camera
        self.player_index = player_index
    @staticmethod
    def read_from(stream):
        window_size = Vec2Int.read_from(stream)
        mouse_pos_window = Vec2Float.read_from(stream)
        mouse_pos_world = Vec2Float.read_from(stream)
        pressed_keys = []
        for _ in range(stream.read_int()):
            pressed_keys_element = stream.read_string()
            pressed_keys.append(pressed_keys_element)
        camera = Camera.read_from(stream)
        player_index = stream.read_int()
        return DebugState(window_size, mouse_pos_window, mouse_pos_world, pressed_keys, camera, player_index)
    def write_to(self, stream):
        self.window_size.write_to(stream)
        self.mouse_pos_window.write_to(stream)
        self.mouse_pos_world.write_to(stream)
        stream.write_int(len(self.pressed_keys))
        for element in self.pressed_keys:
            stream.write_string(element)
        self.camera.write_to(stream)
        stream.write_int(self.player_index)
    def __repr__(self):
        return "DebugState(" + \
            repr(self.window_size) + "," + \
            repr(self.mouse_pos_window) + "," + \
            repr(self.mouse_pos_world) + "," + \
            repr(self.pressed_keys) + "," + \
            repr(self.camera) + "," + \
            repr(self.player_index) + \
            ")"

File: clients/Kotlin-GraalVM-native/src/main/kotlin/model/DebugState.kt
package model

import util.StreamUtil

class DebugState {
    lateinit var windowSize: model.Vec2Int
    lateinit var mousePosWindow: model.Vec2Float
    lateinit var mousePosWorld: model.Vec2Float
    lateinit var pressedKeys: Array<String>
    lateinit var camera: model.Camera
    var playerIndex: Int = 0
    constructor() {}
    constructor(windowSize: model.Vec2Int, mousePosWindow: model.Vec2Float, mousePosWorld: model.Vec2Float, pressedKeys: Array<String>, camera: model.Camera, playerIndex: Int) {
        this.windowSize = windowSize
        this.mousePosWindow = mousePosWindow
        this.mousePosWorld = mousePosWorld
        this.pressedKeys = pressedKeys
        this.camera = camera
        this.playerIndex = playerIndex
    }
    companion object {
        @Throws(java.io.IOException::class)
        fun readFrom(stream: java.io.InputStream): DebugState {
            val result = DebugState()
            result.windowSize = model.Vec2Int.readFrom(stream)
            result.mousePosWindow = model.Vec2Float.readFrom(stream)
            result.mousePosWorld = model.Vec2Float.readFrom(stream)
            result.pressedKeys = Array(StreamUtil.readInt(stream), {
                var pressedKeysValue: String
                pressedKeysValue = StreamUtil.readString(stream)
                pressedKeysValue
            })
            result.camera = model.Camera.readFrom(stream)
            result.playerIndex = StreamUtil.readInt(stream)
            return result
        }
    }
    @Throws(java.io.IOException::class)
    fun writeTo(stream: java.io.OutputStream) {
        windowSize.writeTo(stream)
        mousePosWindow.writeTo(stream)
        mousePosWorld.writeTo(stream)
        StreamUtil.writeInt(stream, pressedKeys.size)
        for (pressedKeysElement in pressedKeys) {
            StreamUtil.writeString(stream, pressedKeysElement)
        }
        camera.writeTo(stream)
        StreamUtil.writeInt(stream, playerIndex)
    }
}
